android-sdk/src/androidTest/java/com/mobilecoin/lib/AccountTest.java

```diff
@@ -99,10 +101,88 @@
         return accountTestData;
     }
 
+    @Test
+    public void test_compare() throws InvalidUriException, SerializationException {
+        Uri fogUri = Uri.parse("fog://some-test-uri");
+        Uri differentFogUri = Uri.parse("fog://some-other-test-uri");
+        Uri fogUriWithPort = Uri.parse("fog://some-test-uri:443");
+        TestFogConfig fogConfig = Environment.getTestFogConfig();
+        RistrettoPrivate key1 = RistrettoPrivate.generateNewKey();
+        RistrettoPrivate key2 = RistrettoPrivate.generateNewKey();
+
+        RistrettoPrivate key1_copy = RistrettoPrivate.fromBytes(key1.getKeyBytes());
+        RistrettoPrivate key2_copy = RistrettoPrivate.fromBytes(key2.getKeyBytes());
+
+        AccountKey first = new AccountKey(
+                key1,
+                key2,
+                fogUri,
+                fogConfig.getFogReportId(),
+                fogConfig.getFogAuthoritySpki()
+        );
+        try {
+            if (first.equals(null)) {
+                Assert.fail("Valid object cannot be equal to null");
+            }
+        } catch (Exception e) {
+            Assert.fail(e.toString());
+        }
+        AccountKey second = new AccountKey(key1_copy,
+                key2_copy,
+                fogUri,
+                fogConfig.getFogReportId(),
+                fogConfig.getFogAuthoritySpki()
+        );
+        if (!first.equals(second)) {
```

Comment:
What's the advantage of writing it this way over "Assert.assertEquals("Public addresses with copied keys must be qual", first, second)"?

Reply:
it's equivalent

Reply:
In that case, consider using "Assert.assertEquals" because it reduces complexity by removing an if statement.

Reply:
done